src/components/App/index.tsx

```diff
@@ -0,0 +1,294 @@
+      swal({
+        button: 'Got It!',
+        content: (
+          <div>
+            <img
+              style={{ width: '175px' }}
+              alt="GT Scheduler Logo"
+              src="/mascot.png"
+            />
+            <h1>GT Scheduler</h1>
+            <p>
+              Hey there, yellow jackets!{' '}
+              <a href="https://github.com/gt-scheduler">GT Scheduler</a> is a
+              new collaboration between{' '}
+              <a href="https://bitsofgood.org/">Bits of Good</a> and{' '}
+              <a href="https://jasonpark.me/">Jason (Jinseo) Park</a> aimed at
+              making class registration easier for everybody! Now, you can
+              access course prerequisites, instructor GPAs, live seating
+              information, and more all in one location.
+            </p>
+            <p>
+              If you enjoy our work and are interested in contributing, feel
+              free to{' '}
+              <a href="https://github.com/gt-scheduler/website/pulls">
+                open a pull request
+              </a>{' '}
+              with your improvements. Thank you and enjoy!
+            </p>
+          </div>
+        )
+      });
+
+      Cookies.set(cookieKey, 'true', { expires: 365 });
+    }
+  }, []);
+
+  // Fetch the current term's scraper information
+  useEffect(() => {
+    setOscar(null);
+    if (term) {
+      axios
+        .get(`https://gt-scheduler.github.io/crawler/${term}.json`)
+        .then((res) => {
+          const newOscar = new Oscar(res.data);
+          setOscar(newOscar);
+        });
+    }
+  }, [term]);
+
+  // Fetch all terms via the GitHub API
+  useEffect(() => {
+    axios
+      .get(
+        'https://api.github.com/repos/gt-scheduler/crawler/contents?ref=gh-pages'
+      )
+      .then((res) => {
+        const newTerms = (res.data as { name: string }[])
+          .map((content) => content.name)
+          .filter((name: string) => /\d{6}\.json/.test(name))
+          .map((name: string) => name.replace(/\.json$/, ''))
+          .sort()
+          .reverse();
+        setTerms(newTerms);
+      });
+  }, [setTerms]);
+
+  // Set the term to be the first one if it is unset
+  // (once the terms load)
+  useEffect(() => {
+    const t = Cookies.get('term');
+    if (!term && (!t || t === 'undefined')) {
+      const [recentTerm] = terms;
+      setTerm(recentTerm);
+    } else if (t && t !== 'undefined') {
+      setTerm(t);
+    }
+  }, [terms, term, setTerm]);
+
+  // Initialize the versionName to Primary
+  useEffect(() => {
+    const v = Cookies.get('version');
+    if (!versionName) {
+      setVersionName(v || 'Primary');
+    }
+  }, [versionName, setVersionName]);
+
+  // Initialize the version lists for each term
+  useEffect(() => {
+    const vs = Cookies.get('versions');
+    if (!vs || vs === '{}') {
+      patchVersionsData(
+        terms.reduce((ac, cur) => ({ ...ac, [cur]: ['Primary', 'New'] }), {})
```

Comment:
'New' definitely shouldn't be its own element in the stored version list since it should just be an additional option. Removing this may simplify other logic as well.

Reply:
I am currently using version lists stored in the cookie as direct inputs into the <Select /> as options. "New" is one of the option and it has an unique icon and onClick passed in as well. We can separate out the "New" from version lists in cookies, although I do not think it will make other implementation any simpler since you would have to manually add the option "New" to the version lists before passing them in as options to <Select /> any way. I do not think it is worth to make this change since it will require changes to many parts and it really does not make other implementation much simpler.

Reply:
I just think that storing "New" as an explicit cookie adds no value (at the very least, you could simply append it to the end whenever you need the full array like `[...arr, "New"]`) and makes the state more brittle. Storing it creates an implicit invariant that it must always exist once at the end of the list for each term, and that invariant could get broken unintentionally in the future (perhaps by future contributors/buggy code). If you think it is justified for the sake of simplicity, I suppose it doesn't hurt and we can leave it in, but it does seem like an anti-pattern (at least to me).

Reply:
I agree. It would probably be counter-intuitive for other people who might be using version lists in the future. I'll try to take 'New' out of the version lists, and append it like you mentioned using spreading. The reason I did not want to modify this was there are other parts that also need to change besides simply appending 'New' as a option; for example, the indexing of addVersion function. I'll go back and make all the corresponding changes.